File: frontend/src/types/calorie.ts
import type { FoodEntry } from './food';

export type LogStatus = 'normal' | 'near_limit' | 'over' | 'under';

export interface CalorieLog {
  id: string;
  log_date: string;
  target_kcal: number;
  consumed_kcal: number;
  remaining_kcal: number;
  status: LogStatus;
  entries: FoodEntry[];
}

export interface CalorieLogSummary {
  log_date: string;
  target_kcal: number;
  consumed_kcal: number;
  remaining_kcal: number;
  status: LogStatus;
}

File: backend/app/schemas/calorie.py
from typing import Optional
from pydantic import BaseModel

from app.schemas.food import FoodEntryResponse


class CalorieLogResponse(BaseModel):
    id: str
    log_date: str
    target_kcal: float
    consumed_kcal: float
    remaining_kcal: float
    status: str
    entries: list[FoodEntryResponse] = []

    class Config:
        from_attributes = True


class CalorieLogSummary(BaseModel):
    log_date: str
    target_kcal: float
    consumed_kcal: float
    remaining_kcal: float
    status: str

    class Config:
        from_attributes = True
